### PR TITLE
[PHP-70] Support for setting the user's address and date of birth

### PR DESCRIPTION
**Pull request by stefandanaita**
This pull requests adds support for setting the user's address and date of birth. It brings the library in line with TrueLayer's current [API reference](https://docs.truelayer.com/reference/create-payment).

### CODE REVIEW

File: src/Entities/Address.php [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        'zip' => 'string|required',
+        'country_code' => 'string|required',
+    ];
+
+    /**
+     * @return string
+     */
+    public function getAddressLine1(): string
+    {
+        return $this->addressLine1;
```
Comment (lighe) [inline src/Entities/Address.php:70]: Could we fallback to `?? null;` for all fields? We don't know if the api will always return a value for them.

File: tests/acceptance/Payment/CreatePayment.php [hunk 2]
```diff
@@ -69,6 +69,26 @@ public function user(): UserInterface
             ->email('alice@truelayer.com');
     }
 
+    public function userWithAddress(): UserInterface
+    {
+        $user = $this->user();
+        $address = $user->addressBuilder()
```
Comment (lighe) [inline tests/acceptance/Payment/CreatePayment.php:75]: Is this method defined anywhere?

File: src/Entities/Address.php [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    ];
+
+    /**
+     * @var string[]
+     */
+    protected array $rules = [
+        'address_line1' => 'string|required',
+        'address_line2' => 'string|nullable',
+        'city' => 'string|required',
+        'state' => 'string|required',
```
Comment (lboynton) [inline src/Entities/Address.php:60]: Is state required? The docs don't say so.

**Review by lboynton — DISMISSED** [review]

File: src/Entities/Address.php (hunk 3 above)
Comment (stefandanaita) [reply src/Entities/Address.php]: Fixed. Good catch! 🙏🏻

**Review by lboynton — DISMISSED** [review]

**Review by lboynton — APPROVED** [review]

**Pull request merged**